modules/Bio/EnsEMBL/Variation/Pipeline/UpdateVariationFeature.pm

```diff
@@ -59,78 +58,124 @@ sub run {
+        while (my @batch = splice(@update_transcripts, 0, 500) ) {
+            my $joined_ids = '"' . join('", "', @batch) . '"';
 
-                $dbc->do(qq{
+            $dbc->do(qq{
                     INSERT IGNORE INTO $temp_table (variation_feature_id, consequence_types)
                     SELECT  variation_feature_id, GROUP_CONCAT(DISTINCT(consequence_types)) 
                     FROM    transcript_variation 
                     WHERE   feature_stable_id IN ($joined_ids)
                     GROUP BY variation_feature_id
                 }) or die "Populating temp table failed";
-
-                @update_transcripts = ();
-            }
         }
 
-        my $joined_ids = '"' . join('", "', @update_transcripts) . '"';
-        next if($joined_ids eq "");
-
-        $dbc->do(qq{
-            INSERT IGNORE INTO $temp_table (variation_feature_id, consequence_types)
-            SELECT  variation_feature_id, GROUP_CONCAT(DISTINCT(consequence_types)) 
-            FROM    transcript_variation 
-            WHERE   feature_stable_id IN ($joined_ids)
-            GROUP BY variation_feature_id
-        }) or die "Populating temp table failed";
+        # Load VF ids that overlap genes determined as deleted, then either set consequence to intergenic if VF has 
+        # no other overlapping genes, or set consequences to those from overlapping non-deleted genes.
+        my @old_genes;
+        my $del_file = $self->param('pipeline_dir') . '/del_log/vf_affected_by_removed_transcripts.txt';
+        if (! -z $del_file) {
+            open (DEL, $del_file) or die "Can't open file $del_file: $!\n";
+            chomp (@old_genes = <DEL>);
+            close(DEL);
+        }
 
-    } else {
+        if(@old_genes) {
+            foreach my $vf_id (@old_genes) {
+
+                my $sth = $dbc->prepare(qq[
+                        SELECT DISTINCT(consequence_types)
+                        FROM transcript_variation
+                        WHERE variation_feature_id = $vf_id
+                     ]);
+                $sth->execute();
+                my $overlap = $sth->fetchall_arrayref();
+                
+                if (defined($overlap->[0]) ) {
+                    my $consequences='';
+                    for my $i (0 .. $#$overlap) {
+                        my $part = $overlap->[$i];
+                        for my $j (0 .. $#$part ) {
+                            $consequences .= $overlap->[$i][$j] . ',';
+                        }
+                    }
+                    my @csqs = split',',$consequences;
+                    my %get_unique = map{$_ => 1 } @csqs;
+                    my @unique_csqs = keys %get_unique;
+                    my $csq_unique = join(",", @unique_csqs);
+
+                    $dbc->do(qq{
+                            INSERT INTO $temp_table (variation_feature_id, consequence_types)
```

Comment:
This step is failing in some cases.
[L101](https://github.com/Ensembl/ensembl-variation/blob/d958315ccb1534131c393ba2fee122c098151564/modules/Bio/EnsEMBL/Variation/Pipeline/UpdateVariationFeature.pm#L101) inserts the new consequence types into the temp table. 
If a gene has updated/new transcripts + deleted transcript this insertion (L144) is going to try to insert the same row into the temp table but it fails due to duplicated keys.
Probably an INSERT IGNORE INTO would be enough to fix the error?